dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -512,4 +542,49 @@ public String toString() {
             return "LazilyMappedRefreshable{" + delegate + '}';
         }
     }
+
+    private static final class InternalDialogueChannelConfiguration {
```

Comment:
probably safest to use immutables here as well, matching other internal types used in this module (e.g. the ChannelCacheKey/etc)